src/web/mjs/connectors/VizShonenJump.mjs

```diff
@@ -34,26 +36,87 @@ export default class VizShonenJump extends Connector {
         };
     }
 
-    async _getMangas() {
+    async _getMangasAvailibleByVolumes() {
+        const request = new Request(new URL('/library', this.url), this.requestOptions);
+        let data = await this.fetchDOM(request);
+
+        data = [...data.querySelectorAll('table.purchase-table a')];
+
+        return data.map(manga => {
+            return {
+                volume_id: manga.pathname,
+                title: manga.innerText.trim(),
+                entry_type: this.EntryTypeEnum.volume
+            };
+        });
+    }
+
+    async _getMangasAvailibleByChapters() {
         const request = new Request(new URL('/shonenjump', this.url), this.requestOptions);
         let data = await this.fetchDOM(request);
 
         if ( !data.innerText.includes('Latest free chapters') ) {
             alert('This website is geolocked. It can only be accessed from the USA.\nYou may use MANGA Plus instead.', this.label, 'info');
             return;
         }
-
-        data = [...data.querySelectorAll('div.o_sort_container div > a')];
+        
+        data = [...data.querySelectorAll('a.o_chapters-link')];
 
         return data.map(manga => {
             return {
-                id: manga.pathname,
-                title: manga.querySelector(':nth-child(2)').innerText.trim()
+                id: JSON.stringify({
+                    id: manga.pathname,
+                    title: manga.innerText.trim(),
+                    entry_type: this.EntryTypeEnum.chapter
+                }),
+                title: manga.innerText.trim(),
             };
         });
     }
 
-    async _getChapters(manga) {
+    async _getMangas() {
+        const mangaAvailibleByVolumes = await this._getMangasAvailibleByVolumes();
+        var mangaAvailibleByChapters = await this._getMangasAvailibleByChapters();
```

Comment:
let is better than var (this goes for every time you used var)

Reply:
Thanks!